### PR TITLE
refactor: groupstypes in driver and assets, address-book detail modal implementing

### PR DESCRIPTION
**Pull request by Daljit-fleetHawks**
## Change Description

I have changed groups functionality in driver and asset form. Implemented user detail modal in address-book- customers/brokers/carriers/vendors, rest users are remaining

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist: (Check only applicable)

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/pages/fleet/assets/add-assets/add-assets.component.ts [hunk 1]
```diff
@@ -304,6 +315,51 @@ export class AddAssetsComponent implements OnInit {
...
+            },
+            error: () => { },
+            next: () => { },
+          });
+      },
+      next: (res) => {
+        this.response = res;
+        this.hasSuccess = true;
+        this.fetchGroups();
+        this.toastr.success('Group added successfully');
```
Comment (kunalfleethawks) [inline src/app/pages/fleet/assets/add-assets/add-assets.component.ts]: Please add period/ fullstop at end of the message.

File: src/app/shared/popups/address-book/address-book.component.ts [hunk 2]
```diff
@@ -291,6 +296,13 @@ export class AddressBookComponent implements OnInit {
     });
   }
 
+  openDetail(targetModal, data) {
+    $('.modal').modal('hide');
+    this.userDetailTitle = data.firstName;
+    const modalRef = this.modalService.open(targetModal);
+    this.userDetailData = data;
+    console.log("this.userDetailData", this.userDetailData);
```
Comment (kunalfleethawks) [inline src/app/shared/popups/address-book/address-book.component.ts]: remove console.log

File: src/app/shared/popups/address-book/address-book.component.ts [hunk 3]
```diff
@@ -438,7 +450,7 @@ export class AddressBookComponent implements OnInit {
   async addBroker() {
     this.hideErrors();
     this.removeUserLocation(this.brokerData.address);
-    let result = await this.checkUserExist(this.brokerData['workPhone'], this.brokerData['workEmail'])
+    // let result = await this.checkUserExist(this.brokerData['workPhone'], this.brokerData['workEmail'])
```
Comment (kunalfleethawks) [inline src/app/shared/popups/address-book/address-book.component.ts]: Remove commented out code

**Review by kunalfleethawks — APPROVED** [review]
@Daljit-fleetHawks please address the comments

File: src/app/shared/popups/address-book/address-book.component.ts (hunk 3 above)
Comment (Daljit-fleetHawks) [reply src/app/shared/popups/address-book/address-book.component.ts]: I need this code...working on it.

**Review by Daljit-fleetHawks — COMMENTED** [review]
Changes done.

**Review by Daljit-fleetHawks — COMMENTED** [review]
changes done

**Review by amritfleethawks — APPROVED** [review]

**Pull request merged**